woocommerce/woocommerce-ios
# Order Editing: Add viewmodel to Edit Address Form

Part of https://github.com/woocommerce/woocommerce-ios/issues/4778, https://github.com/woocommerce/woocommerce-ios/issues/4798

## Description

This PR adds address data prefill on Edit Address Form + unsaved changes check.

### Technical Details

The main addition is `EditAddressFormViewModel` which encapsulates all data logic.
State and Country fields are not added yet, should be merged with `CountrySelectorViewModel`.

### Questions

- Do we need additional "Optional" placeholders for email and phone (they are optional in `Address`), in addition to "Company" and "Address 2"?
- Do we need any validation? Web allows empty data for any of address fields. Maybe remove rest of our "Optional" placeholders then, so user won't be confused that we require all fields except those 2?

(cc @adamzelinski)

## Screenshots

<img width=350 src="https://user-images.githubusercontent.com/3132438/132243754-36894407-778e-4d1d-b21b-476f0d59baea.png">

## Testing

1. Open existing order details and tap edit button for shipping address
2. Observe correct address data in form (except State and Country)
3. Try updating any fields
4. Check that "Done" button in navbar is enabled if there are any unsaved changes

---
Update release notes:
